cmd/kwil-admin/cmds/snapshot/create.go

```diff
@@ -191,32 +193,37 @@ func pgDump(ctx context.Context, dbName, dbUser, dbPass, dbHost, dbPort string,
 	multiWriter := io.MultiWriter(gzipWriter, hasher)
 	var totalBytes int64
 
-	// Pass the output of pg_dump through scanner to sanitize it
-	buf := make([]byte, maxRowSize)
-	scanner := bufio.NewScanner(pgDumpOutput)
-	scanner.Buffer(buf, maxRowSize)
+	// Sanitize the output of pg_dump to include only the necessary tables and data
+	reader := bufio.NewReader(pgDumpOutput)
+
+	for {
+		line, err := reader.ReadString('\n')
```

Comment:
It'll be important that the text dump is newline-terminated, otherwise we'll lose the last line.

Although, didn't we just need to give a very large max to the `scanner.Buffer` call?

Other thing is that I was under the impression that a `bufio.Reader` would also return a `bufio.ErrBufferFull` if it's buffer size was exceeded.  Did we test this behavior?


```go
// NewReader returns a new [Reader] whose buffer has the default size.
func NewReader(rd io.Reader) *Reader {
	return NewReaderSize(rd, defaultBufSize)
}
```

Reply:
Well, nevermind!  It works without raising the buffer size:

```go
func TestXxx(t *testing.T) {
	var b [1000]byte
	b[0] = 2
	b[800] = 1
	b[801] = 2
	b[999] = 1
	r := bytes.NewReader(b[:])

	br := bufio.NewReaderSize(r, 500)
	line, err := br.ReadBytes(1)
	if err != nil {
		t.Fatal(err)
	}
	t.Logf("%x (%d)", line, len(line))
}
```

Reply:
It internally adjusts the buffer sizes.